stattutor/static/js/Initialize_CTATXBlock.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Called by edX to initialize the xblock.
+ * @param runtime - provided by EdX
+ * @param element - provided by EdX
+ */
+function Initialize_CTATXBlock(runtime,element) 
+{
+	var JSONDriver=new X2JS(); // Included library originally obtained at: https://github.com/abdmob/x2js
+	var ll=null;
+	var xblockFrame=document.getElementById("ctatxblockframe");
+	
+    var post = 
+	{
+		save_problem_state: function(state) 
+		{
+			$.ajax({
+			type: "POST",
+			url: runtime.handlerUrl(element, 'ctat_save_problem_state'),
+			data: JSON.stringify({state:state}),
+			contentType: "application/json; charset=utf-8",
+			dataType: "json"});
+		},
+		get_problem_state: function() 
+		{
+			return $.ajax({
+			type: "POST",
+			url: runtime.handlerUrl(element, 'ctat_get_problem_state'),
+			data: JSON.stringify({}),
+			contentType: "application/json; charset=utf-8",
+			dataType: "json"});
+		},
+		report_grade: function(correct_step_count, total_step_count) 
+		{
+			$.ajax(
+			{
+				type: "POST",
+				url: runtime.handlerUrl(element, 'ctat_grade'),
+				data: JSON.stringify({'value': correct_step_count,
+									  'max_value': total_step_count}),
```

Comment:
It looks like you're mixing tabs and spaces? 
This causes the code to display differently in different contexts.

All edX projects use soft tabs (spaces) everywhere; in JavaScript soft tabs should be 4 characters in width.